### PR TITLE
rustbuild: Add more MinGW libraries to ship

### LINKED ISSUE #49044: x86_64-pc-windows-gnu: ld: cannot find -lcredui and -lsecur32 after upgrading to rustc 1.26.0-nightly (521d91c6b 2018-03-14)
https://ci.appveyor.com/project/laumann/compiletest-rs/build/1.0.38/job/o8rkn779qh7ios49

> error: linking with `gcc` failed: exit code: 1
  |
  = note: "gcc" "-Wl,--enable-long-section-names" "-fno-use-linker-plugin" "-Wl,--nxcompat" "-nostdlib" "-m64" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\dllcrt2.o" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsbegin.o" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive0.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive1.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive10.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive11.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive12.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive13.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive14.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive15.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive2.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive3.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive4.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive5.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive6.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive7.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive8.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.serde_derive9.rcgu.o" "-o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.dll" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.crate.metadata.rcgu.o" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\serde_derive-9ebadb45ca9eec02.crate.allocator.rcgu.o" "-Wl,--gc-sections" "-nodefaultlibs" "-L" "C:\\projects\\compiletest-rs\\target\\debug\\deps" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-Wl,-Bstatic" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\libserde_derive_internals-555f0e717120011b.rlib" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\libsyn-9d73a1b0d8b88639.rlib" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\libquote-cce34627b501f031.rlib" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\libproc_macro2-73d79182fd654d90.rlib" "C:\\projects\\compiletest-rs\\target\\debug\\deps\\libunicode_xid-25214f6da763bedd.rlib" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-Wl,-Bdynamic" "-l" "proc_macro-93dbabd980fbabf9" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "syntax-031c8126770b77bb" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "rustc_errors-f2eda81f3e9dcbb7" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "syntax_pos-f46872503870f814" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "rustc_data_structures-c0d56ce5645d455f" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "serialize-60f2dac169f8734f" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "rustc_cratesio_shim-ce4463286aac2a73" "-L" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-l" "std-6b18c18bbc608054" "-Wl,-Bstatic" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcompiler_builtins-362c10b9d34fdd8e.rlib" "-Wl,-Bdynamic" "-l" "kernel32" "-l" "credui" "-l" "setupapi" "-l" "user32" "-l" "secur32" "-l" "advapi32" "-l" "winspool" "-l" "msimg32" "-l" "gdi32" "-l" "opengl32" "-l" "kernel32" "-l" "dbghelp" "-l" "advapi32" "-l" "ws2_32" "-l" "userenv" "-l" "shell32" "-shared" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-lmsvcrt" "-luser32" "-lkernel32" "C:\\Users\\appveyor\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsend.o"
  = note: ld: cannot find -lcredui
          ld: cannot find -lsecur32
>          
> error: aborting due to previous error
> error: Could not compile `serde_derive`.

It builds fine with nightly-2018-03-07

https://ci.appveyor.com/project/laumann/compiletest-rs/build/1.0.38/job/rcfcfv3mfmx7h6t3

The relevant code here: https://github.com/laumann/compiletest-rs/pull/108

### PR DESCRIPTION
**Pull request by alexcrichton**
Closes #49044

### CODE REVIEW

File: src/bootstrap/dist.rs [hunk 1]
```diff
@@ -227,6 +227,8 @@ fn make_win_dist(
         "libwsock32.a",
         "libdbghelp.a",
         "libmsimg32.a",
+        "libcredui.a",
+        "libsecur32.a",
```
Comment (retep998) [inline src/bootstrap/dist.rs]: This list used to be in alphabetical order. I sure do wish it could continue to be in alphabetical order.

**Comment by nikomatsakis** [comment]
r=me once alphabetized :)

**Pull request merged**